dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/dependent/DependentAsyncTaskExecuteFunction.java

```diff
@@ -183,15 +192,30 @@ private List<DependentExecute> initializeDependentTaskList() {
 
     private DependResult calculateDependResult() {
         List<DependResult> dependResultList = new ArrayList<>();
+        Map<String, Long> dependVarPoolEndTimeMap = new HashMap<>();
         for (DependentExecute dependentExecute : dependentTaskList) {
             DependResult dependResult =
                     dependentExecute.getModelDependResult(dependentDate, processInstance.getTestFlag());
+            if (dependResult == DependResult.SUCCESS) {
+                Map<String, Property> varPoolPropertyMap = dependentExecute.getDependTaskVarPoolPropertyMap();
+                Map<String, Long> varPoolEndTimeMap = dependentExecute.getDependTaskVarPoolEndTimeMap();
+                calculateDependVarPool(varPoolPropertyMap, varPoolEndTimeMap, dependVarPoolEndTimeMap);
```

Comment:
Maybe you can write `dependentExecute.addTaskVarPool(...)` here, and it seems you don't have to judge whether `varPoolPropertyMap` is empty.

Reply:
I set this method as static because it needs to be called in both the `DependentExecute` class(line 320) and the `DependentAsyncTaskExecuteFunction` class(line 214), and the calculation of varPool in these two classes has the same logic, just to reuse code.

The judgment here is indeed unnecessary, I will modify it later, thank you.

Reply:
You can make some changes like I said before, then no need to declare static any more.

Reply:
Sorry, it's my fault. I just understood what you said, i will modify it.

Reply:
I have modified the code, thank you for helping me :D